### PR TITLE
Gd test

### PR DESCRIPTION
**Pull request by gdprasadAU**
Refactored AddQuota endpoint.
Removed POST /update-op/:update_operation endpoint.
Changed endpoint URL of AddPlanQuotaDefault to plans/quota-defaults.

### CODE REVIEW

File: internal/controllers/plans.go
Comment (johnworth) [inline internal/controllers/plans.go]: Can we make the JSON key `resource_type_name`? It's more consistent with the naming conventions in our other services.

File: internal/controllers/plans.go
Comment (johnworth) [inline internal/controllers/plans.go]: `Name` should be `name` here.

File: internal/controllers/plans.go
Comment (johnworth) [inline internal/controllers/plans.go]: It should be `resource type` not `resource Type` in the error message.

File: internal/controllers/plans.go
Comment (johnworth) [inline internal/controllers/plans.go]: If we're no longer using this function, then just go ahead and remove it rather than commenting it out. We can always get it out of git's history if we need it.

If the plan is to uncomment it again later, then a comment explaining that would be useful.

File: server/router.go
Comment (johnworth) [inline server/router.go]: 👍

File: server/router.go
Comment (johnworth) [inline server/router.go:56]: 👍

File: server/router.go
Comment (johnworth) [inline server/router.go]: I like the change to the new URL format. It should have a leading `/` on the path, though.

**Review by johnworth — CHANGES_REQUESTED** [review]
Generally good, but there are a few minor changes that should be made before it gets merged.

File: internal/controllers/admins.go
Comment (slr71) [inline internal/controllers/admins.go:68]: 👍

File: internal/controllers/plans.go
Comment (slr71) [inline internal/controllers/plans.go]: The `Find` function that you're using to look up the resource type won't return an error if the resource type can't be found. For that, you'd want to use `First`, `Take`, or `Last`. All of those functions should behave roughly equivalently in this case since there's a uniqueness constraint on the `name` column.

Also, not all errors that come back will indicate that the resource type can't be found. You'll have to check the specific type of error to see if it indicates that the row wasn't found or if it indicates a more severe problem.

You can also use `db.GetResourceTypeByName`. The return value of that function is a little bit different. It returns a pointer to a resource type struct along with an error. If the error is not nil then there's something wrong. If both the resource type pointer and the error are nil, that means that the resource type doesn't exist.

File: internal/controllers/plans.go
Comment (slr71) [inline internal/controllers/plans.go]: The same comment applies here:`Find` won't return an error if the user doesn't exist. You might want to use `First`, `Last`, or `Take` instead.

You could also use `db.GetUser` but the behavior of this endpoint would have to change a little bit if you did. Get user automatically adds new users to the database, so there wouldn't be any error indicating that the username wasn't found.

File: internal/controllers/plans.go
Comment (slr71) [inline internal/controllers/plans.go]: The same comment applies here: `Find` won't return an error if a row isn't found in the database, but `First`, `Last`, and `Take` will. You'll also want to be careful that you're updating the active plan for the user. In this case, I think you'll probably want to call `db.GetActiveUserPlan`; it's probably best not to duplicate the user plan lookup if we can avoid it.

File: internal/controllers/plans.go
Comment (slr71) [inline internal/controllers/plans.go:275]: 👍

**Review by slr71 — CHANGES_REQUESTED** [review]

File: internal/controllers/plans.go
Comment (slr71) [inline internal/controllers/plans.go:247]: `db.GetResourceTypeByName` will still return `nil` for the resource type if a resource type with the given name doesn't exist. It'll be necessary to check for that condition as well.

**Review by slr71 — CHANGES_REQUESTED** [review]
I found one more thing to update.

**Comment by slr71** [comment]
There were some Swagger errors that were preventing the docs from being displayed. I had to take some time troubleshoot the errors, so I fixed them. Normally I wouldn't do that, but I'd already made the changes to verify that they worked while I was testing.

The rest of the changes looked good to me.

**Review by slr71 — APPROVED** [review]

**Comment by slr71** [comment]
At this point, I think we should wait until after Tuesday to merge this PR. That will give us some time to test these changes and any additional changes that we'll have to make.

**Comment by slr71** [comment]
I think this PR is in good shape. I'm going to merge the changes now.

**Pull request merged**
